src/main/java/mekanism/common/inventory/container/ContainerAdvancedElectricMachine.java

```diff
@@ -80,12 +78,7 @@ public ItemStack transferStackInSlot(EntityPlayer player, int slotID) {
     }
 
     private boolean isInputItem(ItemStack itemstack) {
-        for (Entry<AdvancedMachineInput, ItemStack> entry : ((Map<AdvancedMachineInput, ItemStack>) tileEntity.getRecipes()).entrySet()) {
-            if (entry.getKey().itemStack.isItemEqual(itemstack)) {
-                return true;
-            }
-        }
-        return false;
+        return tileEntity.getRecipes().keySet().stream().anyMatch(input -> input.itemStack.isItemEqual(itemstack));
```

Comment:
I'm a little concerned at the overhead of streams here, a for loop on the keyset would be a tiny bit faster with less GC

Is `isItemEqual` the right one here too? or is NBT not an issue?

Reply:
If I remember correctly the JVM is actually pretty good at handling streams efficiently though I can change it if you want.

And I will look into if NBT is an issue or not (I just copied what we had before), though it is certainly possible for all these compares that it was just missing NBT as well.

Reply:
If you can find one of those comparison articles I'm prepared to eat my hat :D

Reply:
Don't get me wrong though, it's not a huge issue

Reply:
Looking at articles it seems that there is a bit of overhead. I think it was when I was first talking with a friend and he was theorizing that streams don't really have much of an overhead. (From what I can tell it seems the main use of them is to simplify parallelism which they have good performance for.) So later I think I will change a couple of these back to just being a simple for loop, and check if you are right about the NBT issue. Given I suspect it probably should check it.